plugins/async-node/ios/Tests/AsynNodePluginTests.swift

```diff
@@ -292,6 +292,126 @@ class AsyncNodePluginTests: XCTestCase {
         XCTAssert(count == 3)
         XCTAssertEqual(expectedMultiNode2Text, "new node from the hook 2")
     }
+
+    func handleEmptyNode() {
+        let context = JSContext()
+        let plugin = AsyncNodePlugin { _ in
+            return .emptyNode
+        }
+        plugin.context = context
+
+        let result = plugin.handleAsyncNodeReplacement(.emptyNode, context: context!)
+        XCTAssertNil(result)
+    }
+
+func handleReplacement(plugin: AsyncNodePlugin, context: JSContext, count: Int) -> AsyncNodeHandlerType {
+    if count == 1 {
+        return .multiNode([
+            ReplacementNode.encodable(AssetPlaceholderNode(asset: PlaceholderNode(id: "text", type: "text", value: "new node from the hook 1"))),
+            ReplacementNode.encodable(AsyncNode(id: "id"))
+        ])
+    } else if count == 2 {
+        return .singleNode(.concrete(context.evaluateScript("""
+            (
+                {"asset": {"id": "text", "type": "text", "value":"new node from the hook 2"}}
+            )
+        """) ?? JSValue()))
+    } else {
+        return .singleNode(.concrete(context.evaluateScript("") ?? JSValue()))
+    }
+}
+
+func testHandleMultipleUpdatesThroughCallback() {
+    let handlerExpectation = XCTestExpectation(description: "first data did not change")
+
+    guard let context = JSContext() else {
+        XCTFail("JSContext initialization failed")
+        return
+    }
+
+    var count = 0
+    var args: [JSValue] = []
+    var callbackFunction: JSValue?
+
+    let resolve: AsyncHookHandler = { callback in
+        handlerExpectation.fulfill()
+        callbackFunction = callback
+
+        let plugin = AsyncNodePlugin { _ in
+            return .singleNode(.concrete(JSValue()))
+        }
+        plugin.context = context
+
+        let replacementResult = self.handleReplacement(plugin: plugin, context: context, count: count)
+        args = [plugin.handleAsyncNodeReplacement(replacementResult, context: context) ?? JSValue()]
+
+        callback.call(withArguments: args)
```

Comment:
shouldnt be invoking the actual `callback`  function in the handler here from this comment https://github.com/player-ui/player/pull/488#discussion_r1720166394 this handler we are in is for the  original resolve, we are just using the handler to assigning the callback for use at a later time to update the original node. 
You can just return a single node with the value "new node from the hook 1" and then for the callback return a single node "new node from the hook 2" which we expect to replace the old value

Reply:
I have removed the callback in the handler and just invoking the callbackFunction after the original node update. Also "**You can just return a single node with the value "new node from the hook" and then for the callback return a single node "new node from the callback" which we expect to replace the old value**" is already being returned by the variable **args** which being passed as an argument to the callback function at a later point